ConduitAPI/Services/Articles/ArticleService.cs

```diff
@@ -0,0 +1,49 @@
+﻿using ConduitAPI.Infrastructure.CommonDto;
+using ConduitAPI.Infrastructure.LinQ;
+using ConduitAPI.Services.Articles.Dto;
+using ConduitAPI.Services.Profile.Dto;
+using Microsoft.EntityFrameworkCore;
+
+namespace ConduitAPI.Services.Articles
+{
+
+    public class ArticleService : IArticleService
+    {
+        private readonly MainDbContext _mainDbContext;
+        public ArticleService(MainDbContext mainDbContext)
+        {
+            _mainDbContext = mainDbContext;
+        }
+        public async Task<PagingResponseDto<ArticleDto>> GetGlobalArticle(QueryGlobalArticleRequestDto request)
+        {
+            var query = _mainDbContext.Articles
+                .WhereIf(!string.IsNullOrEmpty(request.Author), x => x.User.Username == request.Author)
+                .WhereIf(!string.IsNullOrEmpty(request.Favorited), x => x.Favorites.Any(i => i.User.Username == request.Favorited))
+                .WhereIf(!string.IsNullOrEmpty(request.Tag), x => x.Tags.Contains(request.Tag))
+                .OrderByDescending(x => x.CreatedAt);
+            var offset = Math.Max(request.Offset, 1);
```

Comment:
move logic tính toán offset paging vào 1 hàm chung trong file `LinQ/Queryable`
vd
```
public static IQueryable<TSource> Paging<TSource>(this IQueryable<TSource> source, int pageSize, int pageIndex)
        {
            var offset = (Math.Max(pageIndex, 1) - 1) * pageSize;
            return source.Skip(offset).Take(pageSize);
        }
    }
```